integration/setup/setup.go

```diff
@@ -73,15 +73,28 @@ func SetupWithOpts(tb testing.TB, opts *SetupOpts) *SetupResult {
 	logger := testutil.Logger(tb, level)
 
 	var stateProvider *state.Provider
-	var uri string
+	var uri, unixSocketPath string
 	port := *targetPortF
+	targetUnixSocket := *targetUnixSocketF
+
 	if port == 0 {
-		targetUnixSocket := *targetUnixSocketF
-		stateProvider, uri = setupListener(tb, ctx, logger, targetUnixSocket)
-	} else {
-		uri = buildMongoDBURI(tb, port)
+		var socketPath string
+		stateProvider, socketPath, port = setupListener(tb, ctx, logger)
```

Comment:
Can't we just assign to `unixSocketPath` here? 

Like:
```go
stateProvider, unixSocketPath, port = setupListener(tb, ctx, logger)
```

and then we won't need extra checks below.

Reply:
Probably we can't...

Reply:
Then I don't get why we need `-target-unix-socket` flag 😄